tools/mo/openvino/tools/mo/moc_frontend/extractor.py

```diff
@@ -324,6 +330,90 @@ def fe_output_user_data_repack(input_model: InputModel, outputs: list, framework
     return _outputs
 
 
+def convert_params_lists_to_dicts(input_model, input_user_shapes, input_user_data_types, freeze_placeholder, unnamed_freeze_placeholders):
+    from openvino.runtime import PartialShape
+    model_inputs = input_model.get_inputs()
+    freeze_placeholder_dict = freeze_placeholder
+    input_user_data_types_dict = {}
+    input_user_shapes_dict = {}
+    input_user_shapes = input_user_shapes if input_user_shapes is not None else {}
+
+    if isinstance(input_user_shapes, list):
+        for shape in input_user_shapes:
+            assert isinstance(shape, PartialShape), "Got incorrect format of input shapes {}.".format(type(shape))
+        # assert len(model_inputs) == len(input_user_shapes) + len(freeze_placeholder) + \
+        #        len(unnamed_freeze_placeholders), "Got incorrect number of input shapes or frozen values. " \
+        #                                          "Model has {} inputs, got {} input shapes and frozen values. {} {}".format(
+        #     len(model_inputs), len(input_user_shapes) + len(freeze_placeholder) + len(unnamed_freeze_placeholders),
+        #     freeze_placeholder, input_user_shapes
+        # )
+        for idx, shape in enumerate(input_user_shapes):
+            shape_set = False
+            for inp in model_inputs:
+                input_names = inp.get_names()
+                name_found = False
+                for input_name in input_names:
+                    if input_name in freeze_placeholder_dict or input_name in input_user_shapes_dict:
+                        name_found = True
+                        break
+                if name_found:
+                    continue
+                input_user_shapes_dict[input_names[0]] = shape
+                shape_set = True
+                break
+            if not shape_set:
+                raise Error("Could not set shape, as model does not have enough inputs.")
+    else:
+        input_user_shapes_dict = input_user_shapes
+
+    if isinstance(input_user_data_types, list):
+        from openvino.runtime import Type
+        for node_type in input_user_data_types:
+            assert isinstance(node_type, (type, Type)), "Got incorrect format of input types. " \
+                                                        "Expected numpy type or openvino.runtime.Type, " \
+                                                        "got {}.".format(type(node_type))
+
+        for idx, node_type in enumerate(input_user_data_types):
```

Comment:
these 3 search functions have a lot of common (difference only in dicts and error message), maybe we can remove duplicated code and create helper function for them?

Reply:
Added helper method, so looks better now.